packages/constellation/src/components/Compositions/Table/AdvancedTableExample/AdvancedTableExample.stories.tsx

```diff
@@ -0,0 +1,134 @@
+
+export default {
+  argTypes: { onClick: { action: 'clicked' } },
+  args: {},
+  title: 'Compositions/Table/Examples/AdvancedTable',
+} as ComponentMeta<FC>
+
+const Template: ComponentStory<FC<{ onClick: () => void }>> = ({ onClick }) => {
+  const [sorting, setSorting] = React.useState<SortingState>([{ desc: true, id: 'amount' }])
+
+  const columns = React.useMemo<ColumnDef<RowType>[]>(
+    () => [
+      {
+        accessorKey: 'asset',
+        cell: ({ getValue }) => {
+          const props = getValue() as TextCellProps
+          return <TextCell {...props} />
+        },
+        header: ({ header }) => {
+          const sort = header.column.getIsSorted()
+          const onClick = header.column.getToggleSortingHandler()
+
+          return <TextCell text='Asset' isHeader sort={sort} toggleSort={onClick} />
+        },
+        sortingFn: sortTextCells,
+      },
+      {
+        accessorKey: 'amount',
+        cell: ({ getValue }) => {
+          const props = getValue() as TextCellProps
+          return <TextCell {...props} />
+        },
+        header: ({ header }) => {
+          const sort = header.column.getIsSorted()
+          const onClick = header.column.getToggleSortingHandler()
+
+          return <TextCell subtext='Amount' isHeader sort={sort} toggleSort={onClick} />
+        },
+        sortingFn: sortTextCells,
+      },
+      {
+        accessorKey: 'actions',
+        cell: ({ getValue }) => {
+          const props = getValue() as ButtonCellProps
+          return <ButtonCell {...props} />
+        },
+        header: () => <TextCell subtext='Actions' isHeader />,
+      },
+    ],
+    [],
+  )
+
+  const data = [
+    {
+      actions: {
+        primaryButton: { onClick, text: 'Buy', variant: 'primary' },
+        secondaryButton: { onClick, text: 'Sell', variant: 'secondary' },
+      } as ButtonCellProps,
+      amount: { subtext: '1 BTC', text: '$32,000' } as TextCellProps,
+      asset: { subtext: 'BTC', text: 'Bitcoin' } as TextCellProps,
+    },
+    {
+      actions: {
+        primaryButton: { onClick, text: 'Buy', variant: 'primary' },
+        secondaryButton: { onClick, text: 'Sell', variant: 'secondary' },
+      } as ButtonCellProps,
+      amount: { subtext: '1 ETH', text: '$1,800' } as TextCellProps,
+      asset: { subtext: 'ETH', text: 'Ethereum' } as TextCellProps,
+    },
+    {
+      actions: {
+        primaryButton: { onClick, text: 'Buy', variant: 'primary' },
+        secondaryButton: { onClick, text: 'Sell', variant: 'secondary' },
+      } as ButtonCellProps,
+      amount: { subtext: '1 ETH', text: '$1,900' } as TextCellProps,
+      asset: { subtext: 'ETH', text: 'Ethereum' } as TextCellProps,
+    },
+  ]
+
+  const table = useReactTable({
+    columns,
+    data,
+    debugTable: true,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    onSortingChange: setSorting,
+    state: {
+      sorting,
+    },
+  })
```

Comment:
This might be a lot to take in for a new developer trying to build an advanced table.

I don't think we should add to this PR, but what do you think about a follow-up to encapsulate some of this logic into custom hook(s) and/or utils and maybe have a little intro on how to use?

Reply:
Yeah I agree I would like to add some util functions for sorting and possibly simplifying to column defining process. Not sure if there is much we can do about that data but I am open to suggestions!